### PR TITLE
Adding time encoders and decoders for scala 2.12

### PR DESCRIPTION
**Pull request by gpoirier**
This is a Pull Request for the following issue: https://github.com/circe/circe/issues/732

It seems to be the best way to avoid duplication with the cross version support. But it does leave an awkward empty trait in Scala 2.10 and Scala 2.11.

### CODE REVIEW

File: modules/core/shared/src/main/scala-2.10/io.circe/TimeInstances.scala [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package io.circe
+
+trait TimeEncoders {
```
Comment (travisbrown) [inline modules/core/shared/src/main/scala-2.10/io.circe/TimeInstances.scala]: Can we make these traits package-private as well?

Comment (travisbrown) [reply modules/core/shared/src/main/scala-2.10/io.circe/TimeInstances.scala]: Also I'd be tempted to name them `JavaTimeEncoder`, etc., but not a big deal either way.

Comment (gpoirier) [reply modules/core/shared/src/main/scala-2.10/io.circe/TimeInstances.scala]: The only reason the name was `TimeEncoders` was to match what was in the java8 project, I have no preference personally, I'll make the change. But did you omit the "s" intentionally? The other Encoder traits mixed in `Encoder` all have an "s" at the end (`TupleEncoders`, `ProductEncoders`,  `MidPriorityEncoders`).

Comment (travisbrown) [reply modules/core/shared/src/main/scala-2.10/io.circe/TimeInstances.scala]: Thanks, and yes, I meant to include the `s`.

**Comment by gpoirier** [comment]
I can't find the document again with the steps to create a pull request. Was this one good to go, or was it missing something like a rebase or something like that?

**Comment by travisbrown** [comment]
I do still think we want to get the Scala.js issue sorted, and haven't had time to take a closer look at it yet, myself. Otherwise it looks good to me.

**Comment by gpoirier** [comment]
I created a [branch](https://github.com/gpoirier/circe/tree/gpoirier/time-2) with the java time encoders/decoders in the main source file, and moved the tests from the scala-2.12 folder to the scala folder. When running `sbt clean test` locally, it failed. But then I pushed to github, and the travis [build](https://travis-ci.org/gpoirier/circe/builds/279118697) actually passed. I'm not sure why it was failing locally.

**Comment by travisbrown** [comment]
I cherry-picked these commits into #945, so this functionality with be available in 0.10.0-M2.

**Comment by gpoirier** [comment]
That's good new, looking forward to the 0.10.0 release.

**Pull request closed without merging**